asab/library/providers/filesystem.py

```diff
@@ -31,9 +31,22 @@ def __init__(self, library, path, layer, *, set_ready=True):
 		'''
 
 		super().__init__(library, layer)
-		self.BasePath = os.path.abspath(path)
-		while self.BasePath.endswith("/"):
-			self.BasePath = self.BasePath[:-1]
+		# Check for `file://` prefix and strip it if present
+		if path.startswith('file://'):
+			path = path[7:]  # Strip "file://"
+
+		# Handle relative and absolute paths
+		if not os.path.isabs(path):  # Relative path
+			if "../" in path or "./" in path:
+				# Resolve relative paths to absolute
+				path = os.path.abspath(path)
```

Comment:
in both branches (if and else) the same logic is applied. (looking for abs path)